agent/engine/task_manager.go

```diff
@@ -415,8 +415,15 @@ func (task *managedTask) progressContainers() {
 	task.UpdateStatus()
 }
 
-func (task *managedTask) cleanupTask() {
-	cleanupTime := ttime.After(task.KnownStatusTime.Add(taskStoppedDuration).Sub(ttime.Now()))
+func (task *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
+	cleanupTimeDuration := task.KnownStatusTime.Add(taskStoppedDuration).Sub(ttime.Now())
+	// There is a potential deadlock here if cleanupTime is negative. Ignore the computed
+	// value in this case in favor of the default config value.
```

Comment:
My thinking is that I want there to be only two set of values which the cleanupDuration can get. Either the default or a valid value, set by the config. Since a negative value here is a potential bug with an invalid value, i'd prefer it be set to default.